src/lib/y2network/config_reader/sysconfig_dns.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+require "y2network/dns_config"
+
+module Y2Network
+  module ConfigReader
+    # Reads DNS configuration from sysconfig files
+    class SysconfigDNS
```

Comment:
Hmm, to be honest, I do not like much this double name. For me it a bit indicate that maybe we need namespace for ConfigReader::Syconfig and there having stuff like DNS. What do you think?

Reply:
Yes, I prefer that approach too. But I have found sometimes some resistance to have more than 2 levels :) If you are fine, I will move it as I prefer that approach too.

Reply:
well, I am not fan of more then 2 levels, but in this case it makes some sense to me. So as with any rule, use common sense :) Having three levels for few classes is too much for me, but having it for set of class as separator if there are more of such classes, makes sense for me. Another approach can be `Y2Network::ConfigReader::DNS::Sysconfig` where you basically need to use only `ConfigReader::DNS.from_system` to read what is currently used.